link/src/main/java/com/stripe/android/link/LinkActivity.kt

```diff
@@ -3,72 +3,99 @@ package com.stripe.android.link
 import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxHeight
 import androidx.compose.foundation.layout.fillMaxWidth
-import androidx.compose.material.Icon
-import androidx.compose.material.LocalContentColor
-import androidx.compose.material.MaterialTheme
+import androidx.compose.material.CircularProgressIndicator
 import androidx.compose.material.Surface
-import androidx.compose.material.TopAppBar
-import androidx.compose.runtime.Composable
+import androidx.compose.material.Text
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
-import androidx.compose.ui.res.painterResource
-import androidx.compose.ui.res.stringResource
-import androidx.compose.ui.tooling.preview.Preview
+import androidx.lifecycle.viewModelScope
 import androidx.navigation.NavHostController
 import androidx.navigation.compose.NavHost
 import androidx.navigation.compose.composable
 import androidx.navigation.compose.rememberNavController
+import com.stripe.android.link.theme.DefaultLinkTheme
+import com.stripe.android.link.ui.LinkAppBar
 import com.stripe.android.link.ui.signup.SignUpBody
-import com.stripe.android.link.ui.theme.DefaultLinkTheme
+import com.stripe.android.link.ui.verification.VerificationBody
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
 
 @Suppress("EXPERIMENTAL_ANNOTATION_ON_OVERRIDE_WARNING")
 internal class LinkActivity : ComponentActivity() {
-    internal lateinit var navController: NavHostController
+
+    private val viewModel: LinkActivityViewModel by viewModels {
+        LinkActivityViewModel.Factory(
+            application,
+            { requireNotNull(starterArgs) }
+        )
+    }
+
+    @VisibleForTesting
+    lateinit var navController: NavHostController
 
     private val starterArgs: LinkActivityContract.Args? by lazy {
         LinkActivityContract.Args.fromIntent(intent)
     }
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+
         setContent {
             navController = rememberNavController()
+
             DefaultLinkTheme {
                 Surface {
                     Column(Modifier.fillMaxWidth()) {
                         LinkAppBar()
-                        NavHost(navController, startDestination = LinkScreen.SignUp.name) {
-                            composable(LinkScreen.SignUp.name) {
+
+                        NavHost(navController, viewModel.startDestination.route) {
+                            composable(LinkScreen.Loading.route) {
+                                Box(
+                                    modifier = Modifier
+                                        .fillMaxWidth()
+                                        .fillMaxHeight(),
+                                    contentAlignment = Alignment.Center
+                                ) {
+                                    CircularProgressIndicator()
+                                }
```

Comment:
One of the comments we received in a previous version of payment sheet was that our progress indicators were not consistent across the application.  For PaymentSheet the loader is not compose, so it is not super relevant today.  

Just something to keep an eye out for in future development.